### ISSUE TITLE
Visualise disparity?

### ISSUE THREAD

**Issue by Andyjk1406** [gangweiX/ACVNet#13]
I downloaded the sceneflow sample data here https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html

I then pointed the paths to the 3 example images in the FlyingThings3D data set.
I reduced batch size=1, and num_workers=1 in DataLoader (Windows cannot handle multithreading).

I added an inference-only function in main.py and called it directly by changing the final line to 
if __name__ == '__main__':
    AK_Test()

This works and produced the following output:
Epoch 1/64, Iter 0/3, test loss = 0.411, time = 5.176301
Epoch 1/64, Iter 1/3, test loss = 0.528, time = 1.852047
Epoch 1/64, Iter 2/3, test loss = 1.124, time = 1.813151
avg_test_scalars {'loss': 0.6874483029047648, 'EPE': [0.6874483029047648], 'D1': [0.020571456911663216], 'Thres1': [0.04235811159014702], 'Thres2': [0.02612372984488805], 'Thres3': [0.02187207837899526]}

However, I cannot visualise the predicted disparity maps - how is this done? I think they are stored in raw format in the log file?

My inference function in main.py is :

# AK Test
def AK_Test():
    epoch_idx=1
    avg_test_scalars = AverageMeterDict()
    for batch_idx, sample in enumerate(TestImgLoader):    
        global_step = len(TestImgLoader) * 1 + batch_idx
        start_time = time.time()
        #do_summary = global_step % args.summary_freq == 0
        do_summary = 1
        loss, scalar_outputs, image_outputs = test_sample(sample, compute_metrics=do_summary)
        if do_summary:
           save_scalars(logger, 'test', scalar_outputs, global_step)
           save_images(logger, 'test', image_outputs, global_step)
        avg_test_scalars.update(scalar_outputs)
        del scalar_outputs, image_outputs
        print('Epoch {}/{}, Iter {}/{}, test loss = {:.3f}, time = {:3f}'.format(epoch_idx, args.epochs,
                                                                                 batch_idx,
                                                                                 len(TestImgLoader), loss,
                                                                                 time.time() - start_time))
    avg_test_scalars = avg_test_scalars.mean()
    save_scalars(logger, 'fulltest', avg_test_scalars, len(TrainImgLoader) * (epoch_idx + 1))
    print("avg_test_scalars", avg_test_scalars)
    gc.collect()

**Comment by Andyjk1406** [comment]
Hi again - I'll answer my own question here, based on gwcNet repo.
I have modified my inference function as follows, and imported Image from PIL

# AK Test
def AK_Test():
    epoch_idx=1
    avg_test_scalars = AverageMeterDict()
    for batch_idx, sample in enumerate(TestImgLoader):    
        global_step = len(TestImgLoader) * 1 + batch_idx
        start_time = time.time()
        #do_summary = global_step % args.summary_freq == 0
        do_summary = 1
        loss, scalar_outputs, image_outputs = test_sample(sample, compute_metrics=do_summary)
        if do_summary:
           save_scalars(logger, 'test', scalar_outputs, global_step)
           save_images(logger, 'test', image_outputs, global_step)
        avg_test_scalars.update(scalar_outputs)
        
        # Save disparity image
        disp_est_np = tensor2numpy(image_outputs['disp_est'])
        disp_est_npa=np.array(disp_est_np)
        disp_est_npa = np.squeeze(disp_est_npa)
        print(disp_est_npa.shape)  
        print(batch_idx)
        disp_est_uint = np.round(disp_est_npa * 256).astype(np.uint16)
        im = Image.fromarray(disp_est_uint)
        im.save("filename_"+str(batch_idx)+".png")
        
        
        del scalar_outputs, image_outputs
        print('Epoch {}/{}, Iter {}/{}, test loss = {:.3f}, time = {:3f}'.format(epoch_idx, args.epochs,
                                                                                 batch_idx,
                                                                                 len(TestImgLoader), loss,
                                                                                 time.time() - start_time))
    avg_test_scalars = avg_test_scalars.mean()
    save_scalars(logger, 'fulltest', avg_test_scalars, len(TrainImgLoader) * (epoch_idx + 1))
    print("avg_test_scalars", avg_test_scalars)
    gc.collect()
